src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -1010,24 +1010,28 @@ private void WindowClosed(object sender, EventArgs e)
         // passes it to thecurrent workspace
         void DynamoView_KeyDown(object sender, KeyEventArgs e)
         {
-            if (e.Key == Key.Escape)
-                dynamoViewModel.BackgroundPreviewViewModel.CanNavigateBackground = true;
+            if (e.Key == Key.Escape && this.IsMouseOver)
+            {
+                dynamoViewModel.BackgroundPreviewViewModel.NavigationKeyIsDown = true;
+                e.Handled = true;
+            }
         }
 
         void DynamoView_KeyUp(object sender, KeyEventArgs e)
         {
             if (e.Key == Key.Escape && dynamoViewModel.BackgroundPreviewViewModel.CanNavigateBackground)
             {
-                dynamoViewModel.BackgroundPreviewViewModel.CanNavigateBackground = false;
+                dynamoViewModel.BackgroundPreviewViewModel.NavigationKeyIsDown = false;
                 dynamoViewModel.EscapeCommand.Execute(null);
+                e.Handled = true;
```

Comment:
It does not impact the key handling. This will only flag the event as handled if the esc key is down and you're navigating, so it's a pretty constrained condition.